### PR TITLE
[qt5-webengine] Fix issues

### PR DESCRIPTION
**Pull request by Neumann-A**
changes from #16583

 - add feature "proprietary-codecs"
 - fix build error due to internal changes how options are passed.
 - try to fail early on too long build paths instead of late

- [x] wait for CI
- [x] Run `vcpkg x-add-version --all`

### CODE REVIEW

File: ports/qt5-webengine/portfile.cmake [hunk 1]
```diff
@@ -1,7 +1,13 @@
...
-#set(VCPKG_BUILD_TYPE release) #You probably want to set this to reduce build type and space requirements
-message(STATUS "${PORT} requires a lot of free disk space (>300GB), ram (>32 GB) and time (>4h per configuration) to be successfully build.\n\
--- As such ${PORT} is not properly tested.\n\
+string(LENGTH "${CURRENT_BUILDTREES_DIR}" buildtrees_path_length)
+if(buildtrees_path_length GREATER 15 AND CMAKE_HOST_WIN32)
+        message(FATAL_ERROR "${PORT}'s buildsystem uses very long paths and will fail on your system.\n"
+            "We recommend moving vcpkg to a short path such as 'C:\\src\\vcpkg' or using the subst command."
+        )
+endif()
+set(VCPKG_BUILD_TYPE release) #You probably want to set this to reduce build type and space requirements
```
Comment (Neumann-A) [inline ports/qt5-webengine/portfile.cmake:8]:
```suggestion
#set(VCPKG_BUILD_TYPE release) #You probably want to set this to reduce build type and space requirements
```

File: scripts/ci.baseline.txt [hunk 2]
```diff
@@ -1321,7 +1321,7 @@ qt5-webengine:x64-osx=fail
 # D:\buildtrees\qt5-webengine\x64-windows-dbg\src\core\debug\gen\net/third_party/quiche/src/quic/core/proto/cached_network_parameters.pb.h(17):
 # fatal error C1189: #error:  This file was generated by an older version of protoc which is
 # Succesful built requires protobuf to be installed after qt5-webengine not before. Otherwise the build picks up the wrong headers from inside vcpkg.
-qt5-webengine:x64-windows=skip
+# qt5-webengine:x64-windows=skip
```
Comment (Neumann-A) [inline scripts/ci.baseline.txt:1324]:
```suggestion
qt5-webengine:x64-windows=skip
```

**Comment by JackBoosY** [comment]
Seems CI passed.

File: ports/qt5-webengine/portfile.cmake [hunk 3]
```diff
@@ -1,7 +1,13 @@
 vcpkg_check_linkage(ONLY_DYNAMIC_LIBRARY)
+string(LENGTH "${CURRENT_BUILDTREES_DIR}" buildtrees_path_length)
+if(buildtrees_path_length GREATER 35 AND CMAKE_HOST_WIN32)
+        message(FATAL_ERROR "${PORT}'s buildsystem uses very long paths and will fail on your system.\n"
+            "We recommend moving vcpkg to a short path such as 'C:\\vcpkg' or using the subst command."
+        )
+endif()
```
Comment (ras0219-msft) [inline ports/qt5-webengine/portfile.cmake:7]:
```suggestion
vcpkg_buildpath_length_warning(35)
```

**Review by ras0219-msft — CHANGES_REQUESTED** [review]
Thanks for the PR! LGTM once we use the official helper.

If you would like it to be a fatal error, we should rely on the official helper to provide the message and then emit the fatal error afterwards.

File: ports/qt5-webengine/portfile.cmake (hunk 3 above)
Comment (Neumann-A) [reply ports/qt5-webengine/portfile.cmake:7]: The helper is used inside of `qt_submodule_installation`

Comment (Neumann-A) [reply ports/qt5-webengine/portfile.cmake:7]: What I need here is a `vcpkg_buildpath_length_error(35)` and not a warning. Building 2h and getting a long filepath error (file not found although it is there) is not really a good user experience. 
That is why I added it here explicitly.

Comment (ras0219-msft) [reply ports/qt5-webengine/portfile.cmake:7]: Should qt_submodule_installation be modified to terminate then?

Comment (Neumann-A) [reply ports/qt5-webengine/portfile.cmake]: > Should qt_submodule_installation be modified to terminate then?

Most qt modules are fine on longer paths and I think nobody actually knows the correct number to put into the warning. 
In Qt6 for example I only know of qtquick3d having this issue and it builds rather fast compared to qt5-webengine.

**Comment by JackBoosY** [comment]
Depends on #16904

**Review by JackBoosY — APPROVED** [review]

**Pull request merged**